diesel/src/pg/expression/functions.rs

```diff
@@ -2234,3 +2235,52 @@ define_sql_function! {
         values: Arr2
     ) -> Arr2::Out;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// This function row_to_json takes Record type as an Input o convert it to JSON.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     #[cfg(feature = "serde_json")]
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # #[cfg(feature = "serde_json")]
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::sql;
+    /// #     use diesel::sql_types::{Record, Text, Integer};
+    /// #     use serde_json::Value;
+    /// #     let connection = &mut establish_connection();
+    ///
+    ///     // Example SQL query selecting a custom record type
+    ///     let query = diesel::select(sql::<Record<(Text, Integer)>>("ROW('John', 30)::record"));
```

Comment:
Imho, `row_to_json` should be used here instead of `sql` so, the returned value could be casted to json right away.

Reply:
I'm so sorry, will fix this test